includes/class-salesforce.php

```diff
@@ -408,7 +452,10 @@ public static function api_get_salesforce_tokens( $request ) {
 
 		// Must have a valid API key and secret.
 		if ( empty( $settings['client_id'] ) || empty( $settings['client_secret'] ) ) {
-			throw new \Exception( 'Invalid Consumer Key or Secret.' );
+			throw new \WP_Error(
```

Comment:
I don't think we should `throw` at API methods. Instead `return rest_ensure_response( new WP_Error() );`

Reply:
Done in fa51798. 6342ebb2 also adds a missing return if the result of refreshing a token is an error.